x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -9,36 +9,94 @@ import { render } from 'react-dom';
 import { Datasource, DataType } from '..';
 import { DatasourceDimensionPanelProps, DatasourceDataPanelProps } from '../types';
 
-interface IndexPatternPrivateState {
-  query: object;
+type OperationType = 'value' | 'terms' | 'date_histogram';
+
+interface IndexPatternColumn {
+  // Public
+  operationId: string;
+  label: string;
+  dataType: DataType;
+  isBucketed: false;
+
+  // Private
+  operationType: OperationType;
+  arguments: any;
```

Comment:
I know this is subject to change, but can we avoid adding `any` types at all from the start? The arguments can be typed by this construct:
```ts
type OperationType = 'terms' | 'value' | 'date';

interface BaseIndexPatternColumn {
 // public stuff
 operationType: OperationType;
}

interface ValueColumn extends BaseIndexPatternColumn {
 operationType: 'value';
 arguments: {
 }
}

interface TermsColumn extends BaseIndexPatternColumn {
 operationType: 'terms';
 arguments: {
   size:  number;
  // ...
 }
}

type IndexPatternColumn = TermsColumn | ValueColumn;

function fn(x: IndexPatternColumn) {
  if (x.operationType === 'terms') {
    // arguments is known here
  }
}
```

Reply:
I will remove `arguments` here because it's unused in the initial commit.